### PR TITLE
Fix assertion panic for type aliases to anonymous structs

### PR DESCRIPTION
**Pull request by efritz**
Fixes https://github.com/sourcegraph/lsif-go/issues/133

### CODE REVIEW

File: internal/indexer/indexer_test.go [hunk 1]
```diff
@@ -240,49 +240,86 @@ func TestIndexer(t *testing.T) {
 			if value := strings.ReplaceAll(hover.Result.Contents[2].Value, "  ", "\t"); value != expectedUnderlyingType {
 				t.Errorf("incorrect hover text documentation. want=%q have=%q", expectedUnderlyingType, value)
 			}
+		}
```
Comment (efritz) [inline internal/indexer/indexer_test.go:243]: There's actually 3 tests here, should we break these out into different t.Runs?

File: internal/indexer/typestring_test.go [hunk 2]
```diff
@@ -91,6 +91,24 @@ func TestTypeStringEmptyStruct(t *testing.T) {
 	}
 }
 
+func TestTypeStringNameEqualsAnonymousStruct(t *testing.T) {
+	_, f := findDefinitionByName(t, getTestPackages(t), "TestEqualsStruct")
+
+	signature, extra := typeString(f)
+	if signature != "type TestEqualsStruct = struct" {
+		t.Errorf("unexpected type string. want=%q have=%q", "type TestEqualsStruct struct", signature)
```
Comment (efritz) [inline internal/indexer/typestring_test.go:99]: This should fail?

Comment (efritz) [inline internal/indexer/typestring_test.go:99]: Should this fail? I'd expect this to hit typestring.go:50

Comment (Strum355) [reply internal/indexer/typestring_test.go:99]: The tests passed :thinking:

Comment (efritz) [reply internal/indexer/typestring_test.go:99]: Oh the text is wrong here. I'll fix.

Comment (Strum355) [reply internal/indexer/typestring_test.go:99]: Ohh yes cool cool

**Pull request merged**